IM_calculation/scripts/A_KonnoMatricesComputation.py

```diff
@@ -1,47 +1,74 @@
+import argparse
 from pathlib import Path
-import numpy as np
-import os
+from typing import Union
 
+import numpy as np
 from obspy.signal.konnoohmachismoothing import calculate_smoothing_matrix
 
 
-def createKonnoMatrix_single(ft_len):
-    # creates several Konno Ohmachi matrices
+def createKonnoMatrix_single(ft_len: int, bandwidth: int = 20):
+    """
+    Creates a single Konno Ohmachi matrix
+    :param ft_len: Length of Fourier transform
+    :param bandwidth: Bandwidth of Konno Ohmachi smoothing
+    """
     dt = 0.005
     ft_freq = np.fft.rfftfreq(ft_len, dt)
-    konno = calculate_smoothing_matrix(ft_freq, bandwidth=20, normalize=True)
+    konno = calculate_smoothing_matrix(ft_freq, bandwidth=bandwidth, normalize=True)
     return konno
 
 
-def createKonnoMatrices(install_directory, num_to_gen: int = 7):
-    root = os.path.abspath(
-        os.path.join(install_directory, "IM_calculation", "IM", "KO_matrices")
-    )
-    os.makedirs(root, exist_ok=True)
-
+def createKonnoMatrices(
+    install_directory: Union[str, Path], num_to_gen: int = 7, bandwidth: int = 20
+):
+    """
+    Creates several Konno Ohmachi matrices
+    :param install_directory: Directory to install matrices
+    :param num_to_gen: Number of matrices to generate
+    :param bandwidth: Bandwidth of Konno Ohmachi smoothing
+    """
+    Path(install_directory).mkdir(exist_ok=True)
     for i in range(num_to_gen):
         # n = [512, 1024, 2048, 4096, 8192, 16384, 32768]
         n = 512 * 2 ** i
-        file_name = os.path.join(root, f"KO_{n}.npy")
-        matrix = createKonnoMatrix_single(n * 2)
+        file_name = install_directory / f"KO_{n}.npy"
```

Comment:
If install directory is a string this will crash I think?

Reply:
Turns out it never is called with a string anymore so safe to make the typehint a Path instead